2015/presentations/with modern_peripherals: Python and Flask Piotr Dyba/text.md

```diff
@@ -0,0 +1,208 @@
+introduction to Web Sockets, after that I’ll get through implementation, 
+wrappers and an imaginative app for each of the new devices.
+
+##Web Sockets and Python socket solution
+
+In short, web sockets is a protocol providing full-duplex communication channels 
+over a single TCP connection available in web browsers 
+
+![Alt text](./001_web_sockets.png "01 Web Sockets Diagram")
+
+It starts with handshake which upgrades the HTTP connection to use the Web Socket protocol.
+After that there messages exchanges from and too server triggered by user 
+it self or depending on other circumstances defined by programmer – 
+that’s what interests us the most. At the end, one of the sides must send a kill 
+message, which will terminate the connection. 
+Build in Python library is nice but writing JS from scratch on front end would 
+be time consuming so I did not use it. The second solution was SocketIo and its 
+python extension would work well but it requires additional socket only server 
+running which depending on application could be a good solution. Socket.IO enables 
+real-time bidirectional event-based communication. It works on every platform, 
+browser or device. Nevertheless 
+[Flask-SocketIO](https://github.com/miguelgrinberg/Flask-SocketIO)
+was my choice for implementing sockets and serving the 
+application, an all-in-one solution and I really like using the simplicity of flask. 
+The beauty of the @app.route decorators  was applied to that extension, so 
+it is possible to use it in the similar way @socketio.on().
+
+Flask decorator:
+```python
+@app.route('/')
+def index():
+    return render_template('leap_motion.html')
+```
+
+Event in backend activated from frontend and SocketIO decorator:
+```python
+@socketio.on(‘connect’, namespace='/test')
+def disconnect():
+    print('Client connected’)
+```
+
+Event event on frontend activated from backend:
+```python
+def send_msg(msg):
+    emit('msg_event', msg)
+```
+
+The front end is an out-of-the-box solution and usage is as simple as attaching 
+single \<script\> tag library. When using with Flask-SocketIO we have to use 
+0.9.16 version of JS SocketIO library, because of incompatibility with 1.0.
+
+Frontend code:
+```javascript
+$(document).ready(function() {
+    var namespace = '/test';
+    var socket = io.connect('http://' + document.domain + ':' + location.port + namespace);
+    // connecting event
+    socket.on('connect', function () {
+        socket.emit('connect');
+        $('#status').append('connected');
+    });
+    // event on frontend activated from backend
+    socket.on('even_name', function (){
+        console.log("hi");
+        return 0;
+    });
+    // event in backend activated from frontend
+    $('#some_button').click(function() {
+        socket.emit('from_end_event');
+        return 0;
+    });
+});
+```
+
+##Eye Tribe
+
+EyeTribe is eye tracker hardware that is advertised as a cheap solution 
```

Comment:
EyeTribe is an eye tracker hardware that is advertised as a cheap solution. What is eye tracking? It's the process...